osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -44,6 +53,309 @@ const size_t sysfsFlags = IFF_UP | IFF_DEBUG | IFF_NOTRAILERS | IFF_NOARP |
                           IFF_PORTSEL | IFF_AUTOMEDIA | IFF_DYNAMIC;
 #endif
 
+#ifdef __APPLE__
+
+class InterfaceData;
+
+using InteraceDataMap = std::map<std::string, std::shared_ptr<InterfaceData>>;
+
+template <typename Type>
+struct TypeDeleter final {
+  using pointer = Type;
+
+  void operator()(pointer p) {
+    CFRelease(p);
+  }
+};
+
+using UniqueCFStringRef =
+    std::unique_ptr<CFStringRef, TypeDeleter<CFStringRef>>;
+
+using UniqueCFArrayRef = std::unique_ptr<CFArrayRef, TypeDeleter<CFArrayRef>>;
+
+class InterfaceData {
+ public:
+  // Interface name
+  std::string interface_name;
+
+  // Interface type
+  std::string interface_type;
+
+  // Localized Display name
+  std::string display_name;
+
+  // Interface descriptions
+  std::string description;
+
+  // Interface service name
+  std::string service_name;
+
+  // Interface service ID
+  std::string service_id;
+
+  // Interface config method (DHCP/INFIRM/BOOTP/Manual)
+  std::string config_method;
+
+  // DHCP router address
+  std::string router_address;
+
+  // List of DNS servers
+  std::vector<std::string> dns_servers;
+
+  // List of DNS domain names
+  std::vector<std::string> dns_domains;
+
+  // DNS server search order
+  long dns_search_order;
+};
+
+// Trim white spaces from the string
+static inline std::string trim(std::string& str) {
```

Comment:
I think this trims leading spaces? The comment appears off.

In other places we use `boost::algorithm::trim` but :shrug:

Reply:
Comment was updated:

> // Trim leading and trailing white spaces